fix: Update metric endpoint

daniellevass-bitrise opened the pull request:
Updated the metric endpoint and updated the new headers that are required now for this one endpoint.

Tested by making a local trace-sdk and using the pineapple app. I received the following response:
<img width="446" alt="Screenshot 2021-04-07 at 13 11 23" src="https://user-images.githubusercontent.com/71286749/113864469-d50fa300-97a2-11eb-90e7-3f64e6a43be4.png">

richard-bogdan-bitrise commented:
Thanks for adding the response, can you please check the request too?

So we set the headers originally here:
[https://github.com/bitrise-io/trace-android-sdk/blob/main/trace-sdk/src/main/java/io/bitrise/trace/network/NetworkClient.java#L218](https://github.com/bitrise-io/trace-android-sdk/blob/main/trace-sdk/src/main/java/io/bitrise/trace/network/NetworkClient.java#L218)

and according to the javadoc of Headers annotation, headers with the same name are not overwritten, but a new one will be added:
[https://square.github.io/retrofit/2.x/retrofit/](https://square.github.io/retrofit/2.x/retrofit/)

To avoid confusions, I would remove these two header fields from NetworkClient, and manually add to sendTraces in NetworkCommunicator the Accept and Content type fields. So these two fields will be in the same class, both for metrics and traces. Might consider to add this information to the javadocs. Please let me know your thoughts about this.

daniellevass-bitrise commented:
> Thanks for adding the response, can you please check the request too?
> 
> So we set the headers originally here:
> https://github.com/bitrise-io/trace-android-sdk/blob/main/trace-sdk/src/main/java/io/bitrise/trace/network/NetworkClient.java#L218
> 
> and according to the javadoc of Headers annotation, headers with the same name are not overwritten, but a new one will be added:
> https://square.github.io/retrofit/2.x/retrofit/
> 
> To avoid confusions, I would remove these two header fields from NetworkClient, and manually add to sendTraces in NetworkCommunicator the Accept and Content type fields. So these two fields will be in the same class, both for metrics and traces. Might consider to add this information to the javadocs. Please let me know your thoughts about this.

Thanks for the feedback Richard!

I removed the headers from the NetworkClient (and alphabeticalised the list) and moved them into the NetworkConnector. I also added some javadoc to explain which was the updated endpoint, and that hopefully we can make the traces endpoint updated at some point. Does that look okay to you?
